inference-engine/src/inference_engine/include/ie/ie_data.h

```diff
@@ -142,6 +167,18 @@ class INFERENCE_ENGINE_API_CLASS(Data) {
      */
     const UserValue& getUserObject() const;
 
+    /**
+     * @brief Checks if current data has dynamic shapes
+     * @return true if data has dynamic shapes
+     */
+    bool isDynamic() const;
+
+    /**
+     * @brief Returns partial shapes
+     * @return shapes which can have dynamic dimensions
+     */
+    const ngraph::PartialShape& getPartialShape() const;
```

Comment:
Should we return an instance of pyngraph::PartialShape or convert it to python list in IE PYTHON API?

Reply:
We should have _good_ PartialShape implementation in Python API. And this doesn't look as a list, because PartialShape has other methods that list doesn't have. Whether PartialShape already have Python API implementation.

Reply:
Is it a question about current C++ API or about future python implementation?